### ISSUE TITLE
clang-analyzer.UndefReturn false positive

### ISSUE THREAD

**Issue by briangreenery** [llvm/llvm-project#55252]
I think I have run into a false positive with `clang-analyzer-core.uninitialized.UndefReturn`. Here is the code that it warns on. It only warns when compiled with `-stdlib=libc++` on clang 14.

A repro is also here: https://godbolt.org/z/46q734cM3

```
#include <algorithm>

class IntRef
{
public:
    explicit IntRef(int& x) : m_p(&x) {}
        
    int& get() const { return *m_p; }

    int* m_p;
};

bool operator<(IntRef a, IntRef b) {
    return *a.m_p < *b.m_p;
}

void foo(IntRef* values, int size) {
    std::stable_sort(values, values + size);
}
```

And it warns:

```
[warning: Undefined or garbage value returned to caller [clang-analyzer-core.uninitialized.UndefReturn]]
    int& get() const { return *m_p; }
                       ^
```

It seems like small changes to the code (like changing `get()` to return `int` instead of `int&`) resolve the warning.

**Comment by martong** [comment]
The problem is that C++ member functions are being analyzed as top-level functions. I.e. in this case, the constructor is not analyzed (simulated) before the call of `get`.

@haoNoQ Perhaps we should have a flag that would be more conservative and would disable the top-level analysis of member functions? There must be a created object to call a non-static member function anyway.
